auth0_flutter/lib/src/web/jwt_decode.dart

```diff
@@ -0,0 +1,28 @@
+import 'dart:convert';
+
+import 'jwt_decode_exception.dart';
+
+class JWT {
+  static Map<String, dynamic> decode(final String jwt) {
+    final parts = jwt.split('.');
+
+    if (parts.length != 3) {
+      throw JWTDecodeException.invalidPartCount(jwt, parts.length);
+    }
+
+    final String stringPayload;
+
+    try {
+      final decodedPayload = base64.decode(base64.normalize(parts[1]));
+      stringPayload = utf8.decode(decodedPayload);
+    } catch (error) {
+      throw JWTDecodeException.invalidBase64URL(parts[1]);
+    }
+
+    try {
+      return jsonDecode(stringPayload) as Map<String, dynamic>;
+    } catch (error) {
+      throw JWTDecodeException.invalidJSON(parts[1]);
+    }
+  }
```

Comment:
We can already get the decoded token from the SDK using `getIdTokenClaims` - do we need to decode it manually here? 

https://auth0.github.io/auth0-spa-js/classes/Auth0Client.html#getIdTokenClaims

Reply:
Not sure we can represent custom claims (https://github.com/auth0/auth0-spa-js/blob/master/src/global.ts#L578) without using a `Map<String, dynamic>`, hence the custom decoding.